src/drawers/drawer.ts

```diff
@@ -76,7 +76,14 @@ export class Drawer {
    * @param{any[]} data The data to be drawn
    */
   private _bindSelectionData(data: any[]) {
-    let dataElements = this.selection().data(data);
+    // if the dataset has a key, use it when binding the data   
+    let dataElements: d3.selection.Update<any>;
+    if (this._dataset && this._dataset.keyFunction()) {
+      dataElements = this.selection().data(data, this._dataset.keyFunction());
+    } else {
+      dataElements = this.selection().data(data);
+    }
```

Comment:
What probably happened is that the tests explicitly passed in `null`.  Let's leave this in for now then.  I can go look for a fix on my end.